### PR TITLE
[trainer] feat: support BAGEL non-lora training with pickscore reward

### PR DESCRIPTION
**Pull request by zhtmike**
### What does this PR do?
Support bagel non-lora training.

- Add `run_bagel_pickscore.sh` for BAGEL non-LoRA FlowGRPO training
- Add `DiffusionModelBase.configure_trainable_params` hook for non-LoRA `requires_grad` control
- Freeze understanding pathway, train `moe_gen` only, cast to fp32
- Call `configure_trainable_params` in engine when `lora_rank=0`, before FSDP wrapping
- Route actor-synced `transformer.*` weights through `language_model.load_weights` for qkv fusion
- Fix RoPE `cos`/`sin` dtype mismatch
- Fix `NonDiffusersModelBase` metaclass to explicit `ABCMeta`

> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `vllm_omni`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `diffusion`, `omni`, `tests`, `docker`
  - If this PR involves multiple modules, separate them with `,` like `[diffusion, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][diffusion, fsdp] feat: new rollout scheduler`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

螢幕截圖 2026-07-06 上午9 35 42

螢幕截圖 2026-07-06 上午9 35 21

螢幕截圖 2026-07-06 上午9 35 05

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update the documentation.
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl-omni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

**Comment by zhtmike** [comment]
code is ready to review; the wandb result is at https://wandb.ai/mikecheung/flow_grpo/runs/ybv9jf3y, I will update the PR's content once it nearly finish training

**Review by AndyZhou952 — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]
LGTM

**Comment by zhtmike** [comment]
there is a slightly model degration compared with lora, let me check first
